src/transpose.jl

```diff
@@ -52,54 +52,52 @@ Base.:(==)(A::LinearMap, B::AdjointMap)      = ishermitian(A) && B.lmap == A
-end
-function _unsafe_mul!(y::AbstractVecOrMat, A::TransposeMap, x::AbstractVector, α::Number, β::Number)
-    issymmetric(A.lmap) ? _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapvec_mul!(y, A, x, α, β)
-end
-function _unsafe_mul!(y::AbstractMatrix, A::TransposeMap, x::AbstractMatrix, α::Number, β::Number)
-    issymmetric(A.lmap) ? _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapmat_mul!(y, A, x, α, β)
-end
+_unsafe_mul!(y::AbstractVecOrMat, A::TransposeMap, x::AbstractVector) =
+    issymmetric(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x) : error("transpose not implemented for $(A.lmap)")
+_unsafe_mul!(y::AbstractMatrix, A::TransposeMap, x::AbstractMatrix) =
+    issymmetric(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x) : _generic_mapmat_mul!(y, A, x)
+_unsafe_mul!(y::AbstractVecOrMat, A::TransposeMap, x::AbstractVector, α::Number, β::Number)=
+    issymmetric(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapvec_mul!(y, A, x, α, β)
+_unsafe_mul!(y::AbstractMatrix, A::TransposeMap, x::AbstractMatrix, α::Number, β::Number) =
+    issymmetric(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapmat_mul!(y, A, x, α, β)
 # # AdjointMap
-function _unsafe_mul!(y::AbstractVecOrMat, A::AdjointMap, x::AbstractVector)
-    ishermitian(A.lmap) ? _unsafe_mul!(y, A.lmap, x) : error("adjoint not implemented for $(A.lmap)")
-end
-function _unsafe_mul!(y::AbstractMatrix, A::AdjointMap, x::AbstractMatrix)
-    ishermitian(A.lmap) ? _unsafe_mul!(y, A.lmap, x) : _generic_mapmat_mul!(y, A, x)
-end
-function _unsafe_mul!(y::AbstractVecOrMat, A::AdjointMap, x::AbstractVector, α::Number, β::Number)
-    ishermitian(A.lmap) ? _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapvec_mul!(y, A, x, α, β)
-end
-function _unsafe_mul!(y::AbstractMatrix, A::AdjointMap, x::AbstractMatrix, α::Number, β::Number)
-    ishermitian(A.lmap) ? _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapmat_mul!(y, A, x, α, β)
-end
+_unsafe_mul!(y::AbstractVecOrMat, A::AdjointMap, x::AbstractVector) =
+    ishermitian(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x) : error("adjoint not implemented for $(A.lmap)")
+_unsafe_mul!(y::AbstractMatrix, A::AdjointMap, x::AbstractMatrix) =
+    ishermitian(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x) : _generic_mapmat_mul!(y, A, x)
+_unsafe_mul!(y::AbstractVecOrMat, A::AdjointMap, x::AbstractVector, α::Number, β::Number) =
+    ishermitian(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapvec_mul!(y, A, x, α, β)
+_unsafe_mul!(y::AbstractMatrix, A::AdjointMap, x::AbstractMatrix, α::Number, β::Number) =
+    ishermitian(A.lmap) ?
+        _unsafe_mul!(y, A.lmap, x, α, β) : _generic_mapmat_mul!(y, A, x, α, β)
+
 # # ConjugateMap
-for (intype, outtype) in ((AbstractVector, AbstractVecOrMat), (AbstractMatrix, AbstractMatrix))
+const ConjugateMap = AdjointMap{<:Any, <:TransposeMap}
+# canonical order of adjoint followed by transpose
+LinearAlgebra.transpose(A::AdjointMap) = adjoint(transpose(A.lmap))
```

Comment:
Maybe add
```suggestion
LinearAlgebra.transpose(A::AdjointMap) = adjoint(transpose(A.lmap))
LinearAlgebra.transpose(A::ConjugateMap) = adjoint(A.lmap.lmap)
```
to keep the types short?

Reply:
Yes good idea